controllers/hardwareclassificationcontroller_controller.go

```diff
@@ -35,6 +35,8 @@ type HardwareClassificationControllerReconciler struct {
 
 // +kubebuilder:rbac:groups=metal3.io.sigs.k8s.io,resources=hardwareclassificationcontrollers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=metal3.io.sigs.k8s.io,resources=hardwareclassificationcontrollers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts,verbs=create;delete;get;list;patch;update;watch
+// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts/status,verbs=get;patch;update
```

Comment:
This should be restricted to the bare minimum needed for it to work. You should probably not have verbs like create and delete there, and you probably do not need patch and update on the status. Once modified, please run kubebuilder to generate the rbac role properly and verify that it has only the required permissions

Reply:
you might be missing something btw in your manifests target in your makefile for the rbac role generation to work properly

Reply:
+1

Reply:
In next commit I have kept bare minimum verbs also ran kubebuilder to generate rbac role properly.